### PR TITLE
Feature/talks resources

### PR DESCRIPTION
**Pull request by JoeJasinski**
@alysivji @tathagata I've gotten many of the API endpoints written. The /vote/ endpoint still needs some work.

### CODE REVIEW

File: talkvoter/settings.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import os
 
+SECRET_KEY = os.getenv("SECRET_KEY")
 
 SQLALCHEMY_DATABASE_URI = os.getenv("DATABASE_URL", "sqlite:///:memory:")
 
 SQLALCHEMY_TRACK_MODIFICATIONS = False
+
+ERROR_404_HELP = False
```
Comment (alysivji) [inline talkvoter/settings.py]: Flask has a few config idioms.

https://github.com/alysivji/flask-mega-tutorial/blob/master/app/config.py to create
https://github.com/alysivji/flask-mega-tutorial/blob/master/app/__init__.py to use

File: talkvoter/v1_resources.py [hunk 2]
```diff
@@ -1,22 +1,70 @@
...
+from .models import db, Talk
+from .serializers import VoteSchema, TalkSchema
 from .constants import VoteValue
 
 
 api_bp = Blueprint('api_v1', __name__)
 api = Api(api_bp)
 
 
+class TalksResource(Resource):
```
Comment (alysivji) [inline talkvoter/v1_resources.py]: Very similar name to `TalkResources`, maybe use `TalksListResource`

File: talkvoter/v1_resources.py [hunk 3]
```diff
@@ -1,22 +1,70 @@
...
+
+api.add_resource(TalksResource, '/talks/', endpoint="api.talks")
+
+
+class TalkResource(Resource):
+
+    def get(self, id):
+        schema = TalkSchema()
+        data = {}
+        talk_obj = db.session.query(Talk).filter(Talk.id == id).first()
```
Comment (alysivji) [inline talkvoter/v1_resources.py]: There is a flask plugin for SQL alchemy that has *_or_404() helpers

http://flask-sqlalchemy.pocoo.org/2.3/api/

This works so I'm good either way.

File: talkvoter/models.py [hunk 4]
```diff
@@ -29,6 +30,8 @@ class Vote(db.Model):
     __tablename__ = 'vote'
 
     id = db.Column(db.Integer, primary_key=True)
+    talk_id = db.Column(db.Integer, db.ForeignKey('talk.id'))
+    author = relationship("Talk", backref="vote")
```
[the remark this answers is not in the document]
Comment (JoeJasinski) [reply talkvoter/models.py]: I think that 'author' was a result of a copy/paste.

I haven'r added the user model yet, but that's next

File: talkvoter/settings.py (hunk 1 above)
Comment (JoeJasinski) [reply talkvoter/settings.py]: updated config

File: talkvoter/v1_resources.py (hunk 2 above)
Comment (JoeJasinski) [reply talkvoter/v1_resources.py]: makes sense... changed

File: talkvoter/v1_resources.py (hunk 3 above)
Comment (JoeJasinski) [reply talkvoter/v1_resources.py]: Interesting, I functionalized this logic, and then I saw your comment.  Do you know if that _or_404 method can take a custom error message?

Comment (alysivji) [reply talkvoter/v1_resources.py]: Not that I recall. Think it just bubbles up a `NotFound` exception which Flask turns into a 404.

I like the flexibility of having more info and honestly, like most flask plugins, it's saving us from writing 10-20 lines of code. This looks good to me!

File: talkvoter/models.py [hunk 5]
```diff
@@ -29,6 +30,8 @@ class Vote(db.Model):
     __tablename__ = 'vote'
 
     id = db.Column(db.Integer, primary_key=True)
+    talk_id = db.Column(db.Integer, db.ForeignKey('talk.id'))
+    talk = relationship("Talk", backref="vote")
```
Comment (alysivji) [inline talkvoter/models.py]: Would change to `backref='votes'` since it'll be a one to many

File: talkvoter/serializers.py [hunk 6]
```diff
@@ -8,3 +8,10 @@ class Meta:
         jit = toastedmarshmallow.Jit
         model = Vote
         fields = ['value']
+
+
+class TalkSchema(ModelSchema):
+    class Meta:
+        jit = toastedmarshmallow.Jit
```
Comment (alysivji) [inline talkvoter/serializers.py]: :D

**Pull request merged**